core/src/main/java/org/apache/iceberg/inmemory/InMemoryCatalog.java

```diff
@@ -88,11 +96,16 @@ protected TableOperations newTableOps(TableIdentifier tableIdentifier) {
   }
 
   @Override
-  protected String defaultWarehouseLocation(TableIdentifier tableIdentifier) {
+  public String defaultWarehouseLocation(TableIdentifier tableIdentifier) {
```

Comment:
@rdblue given that `BaseViewCatalog` is an interface, the downside is that we need to make this method public. 
The alternative would be: we could make `BaseViewCatalog` an abstract class that extends `BaseMetastoreCatalog`. Then the `InMemoryCatalog` would extend `BaseViewCatalog` instead of `BaseMetastoreCatalog`.